include/ofi_coll.h

```diff
@@ -67,6 +67,7 @@ struct util_av_set {
 	fi_addr_t		*fi_addr_array;
 	size_t			fi_addr_count;
 	uint64_t		flags;
+	struct util_coll_mc     *coll_mc;
```

Comment:
We can probably just embed the structure, rather than storing a pointer and needing to alloc/free it.

Doing that, util_coll_mc_alloc() should be renamed to util_coll_mc_init(), with the calloc pulled out.  The caller (only place would be fi_join_multicast) would allocate the structure, then call init() to initialize it.

Reply:
done